mithril-aggregator/src/signer_registration/slave.rs

```diff
@@ -0,0 +1,345 @@
+        "Signer registration"
+    }
+
+    async fn prune(&self, epoch: Epoch) -> StdResult<()> {
+        let registration_epoch = epoch.offset_to_recording_epoch();
+
+        if let Some(retention_limit) = self.verification_key_epoch_retention_limit {
+            self.verification_key_store
+                .prune_verification_keys(registration_epoch - retention_limit)
+                .await
+                .with_context(|| {
+                    format!(
+                        "VerificationKeyStorer can not prune verification keys below epoch: '{}'",
+                        registration_epoch - retention_limit
+                    )
+                })?;
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use anyhow::anyhow;
+    use mockall::predicate::eq;
+
+    use mithril_common::{
+        entities::{Epoch, Signer, SignerWithStake},
+        test_utils::MithrilFixtureBuilder,
+    };
+
+    use crate::{
+        database::{repository::SignerRegistrationStore, test_helper::main_db_connection},
+        services::EpochPruningTask,
+        signer_registration::{
+            MockSignerRecorder, MockSignerRegistrationVerifier, SignerSynchronizer,
+        },
+        store::MockVerificationKeyStorer,
+        MithrilSignerRegistrationSlave, SignerRegisterer, SignerRegistrationRoundOpener,
+    };
+
+    #[tokio::test]
+    async fn open_registration_always_succeeds() {
+        let verification_key_store = Arc::new(SignerRegistrationStore::new(Arc::new(
+            main_db_connection().unwrap(),
+        )));
+        let signer_registerer = MithrilSignerRegistrationSlave::new(
+            verification_key_store.clone(),
+            Arc::new(MockSignerRecorder::new()),
+            Arc::new(MockSignerRegistrationVerifier::new()),
+            None,
+        );
+        let registration_epoch = Epoch(1);
+        let fixture = MithrilFixtureBuilder::default().with_signers(1).build();
+        let stake_distribution = fixture.stake_distribution();
+
+        signer_registerer
+            .open_registration_round(registration_epoch, stake_distribution)
+            .await
+            .expect("signer registration round opening should not fail");
+    }
+
+    #[tokio::test]
+    async fn close_registration_always_succeeds() {
+        let verification_key_store = Arc::new(SignerRegistrationStore::new(Arc::new(
+            main_db_connection().unwrap(),
+        )));
+        let signer_registerer = MithrilSignerRegistrationSlave::new(
+            verification_key_store.clone(),
+            Arc::new(MockSignerRecorder::new()),
+            Arc::new(MockSignerRegistrationVerifier::new()),
+            None,
+        );
+
+        signer_registerer
+            .close_registration_round()
+            .await
+            .expect("signer registration round opening should not fail");
+    }
+
+    #[tokio::test]
+    async fn register_signer_always_fails() {
```

Comment:
suggestion: `register_signer_always_fails_on_slave`

Reply:
The test is in the `slave` module so suffixing with `slave` is not much more explicit.